### PR TITLE
wxmaxima: 22.05.0 -> 22.12.0

### LINKED ISSUE #210224: wxmaxima out of date
I tried to update `wxmaxima` myself, but simply bumping the version number leads to the fallowing error:

```
-- Looking for wxWidgets in:           /nix/store/fnz241dg38svrdgqjxfxv8d68izbf719-wxwidgets-3.2.1
-- Looking for wxWidgets libraries in: 
CMake Error at /nix/store/blqlkgmvs12h8m8rhbv7pmyj7jyxxp37-cmake-3.24.3/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find wxWidgets (missing: wxWidgets_LIBRARIES) (Required is at
  least version "3")
Call Stack (most recent call first):
  /nix/store/blqlkgmvs12h8m8rhbv7pmyj7jyxxp37-cmake-3.24.3/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/blqlkgmvs12h8m8rhbv7pmyj7jyxxp37-cmake-3.24.3/share/cmake-3.24/Modules/FindwxWidgets.cmake:977 (find_package_handle_standard_args)
  CMakeLists.txt:137 (find_package)


-- Configuring incomplete, errors occurred!
```

###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: [wxmaxima](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/math/wxmaxima/default.nix)
current version: [22.05](https://github.com/wxMaxima-developers/wxmaxima/releases/tag/Version-22.05.0) 
desired version: [22.12](https://github.com/wxMaxima-developers/wxmaxima/releases/tag/Version-22.12.0)

###### Notify maintainers
maintainers: @doronbehar 

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Closes #210224.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
> 1. Trying to open or save files crashes the application
>    ```
>    (wxmaxima:196949): GLib-GIO-ERROR **: 18:51:23.726: No GSettings schemas are installed on the system
>    ```
>      
>    This is fixed by setting `XDG_DATA_DIRS` as proposed in [appimage-run: GLib-GIO-ERROR**: No GSettings schemas are installed on the system #72282](https://github.com/NixOS/nixpkgs/issues/72282).

Thanks! I am able to reproduce the first issue under Sway. Adding `glib` to buildInputs should also resolve it, and seems less hacky to me. I don't have the second issue on my machine, though.

**Comment by kilianar** [comment]
Adding `glib`to `buildInputs` resolves the first issue and is indeed the straightforward solution.

As you can't reproduce the second issue it's probably a problem with my theming related configuration. Unfortunately I don't have time to investigate the problem thoroughly right now.

**Comment by doronbehar** [comment]
> Adding `glib`to `buildInputs` resolves the first issue and is indeed the straightforward solution.

Can you do this please @wegank ?

**Comment by wegank** [comment]
Done.

File: pkgs/applications/science/math/wxmaxima/default.nix [hunk 1]
```diff
@@ -11,20 +11,22 @@
...
+    sha256 = "sha256-RT6y4M6LQD1fXJcjtdSXnDmoJvv160g2asdV4WtTcok=";
   };
 
   buildInputs = [
     wxGTK
     maxima
     # So it won't embed svg files into headers.
     gnome.adwaita-icon-theme
+    # So it won't crash under Sway.
+    glib
```
Comment (kilianar) [inline pkgs/applications/science/math/wxmaxima/default.nix:29]: `glib` is missing as an argument, causing the build to fail.

Comment (wegank) [reply pkgs/applications/science/math/wxmaxima/default.nix:30]: Oh my! Fixed.

**Review by kilianar — APPROVED** [review]
LGTM now. Thanks again for your efforts!

**Pull request merged**